packages/vite-plugin-svelte/src/utils/dependencies.ts

```diff
@@ -187,12 +187,23 @@ export function needsOptimization(dep: string, localRequire: NodeRequire): boole
 	const pkg = depData.pkg;
 	// only optimize if is cjs, using the below as heuristic
 	// see https://github.com/sveltejs/vite-plugin-svelte/issues/162
-	const isCjs = pkg.main && !pkg.module && !pkg.exports;
-	if (!isCjs) return false;
-	// ensure entry is js so vite can prebundle it
-	// see https://github.com/sveltejs/vite-plugin-svelte/issues/233
-	const entryExt = path.extname(pkg.main);
-	return !entryExt || entryExt === '.js' || entryExt === '.cjs';
+	if (pkg.type !== 'module' && !pkg.module) {
```

Comment:
the pkg.module check here has been kept from the previous heuristic. It assumes that vite will resolve via that module field.

Reply:
also, @bluwy pointed out to me the other day that even if `pkg.type === 'module'` you can still have a `.cjs` in the `main` and so we don't know that it's ESM. I've never seen that in the wild, but it's technically possible

Reply:
> If `exports` is defined, I think that will take precedence over `pkg.module` though?

module is only ever used by bundlers, so it would be up to them. The presence is a strong hint that it supports esm when bundled. I think vite honors module but have to check

Reply:
Vite does use `module`. I suppose it's fair to say that if `module` is defined then it supports ESM